### PR TITLE
#151: organization.schemes does not reflect current state of schemes...

### PR DESCRIPTION
**Pull request by dkent600**
#151: organization.schemes does not reflect current state of scheme registration

Improvements to type definitions (arc.d.ts)

Added an automated test for #140 UpgradingScheme removes itself when its own address is passed as new UpgradingScheme

### CODE REVIEW

**Comment by leviadam** [comment]
Hi @dkent600,

I'm not sure about using the wrappers in the tests.
Rethinking, maybe to take the library to a different repo.
What do you think @orenyodfat @tibetsprague ?

**Comment by orenyodfat** [comment]
I support for having two separate repositories  : 
1. Solidity contracts + contracts tests.
2. JS library  + tests.

I guess it will be easier to maintain and stabilize versions of daostack stack's components.

**Comment by leviadam** [comment]
OK, so it is decided. We go with two repos. So I close this PR. Move things to the new repo. And @dkent600 will PR to that repo.

File: dist/arc.js [hunk 1]
```diff
@@ -1,115 +1,115 @@
 'use strict';
 
 Object.defineProperty(exports, "__esModule", {
-  value: true
+    value: true
```
Comment (tibetsprague) [inline dist/arc.js]: i prefer 2 spaces myself

Comment (leviadam) [reply dist/arc.js]: LOL. We just moved from 2 to 4 in the solidity :)

Comment (dkent600) [reply dist/arc.js]: hmm, that is I set lint to -- 2.

Comment (dkent600) [reply dist/arc.js]: @tibetsprague You should look at the code in the PR in arc-js, not here.

Comment (dkent600) [reply dist/arc.js]: This PR is closed and the code changes obsolete, at least with respect to lint-related improvements.

Comment (dkent600) [reply dist/arc.js]: @tibetsprague  Do you recommend 4 then, or 2?  I don't have a strong preference.

Comment (tibetsprague) [reply dist/arc.js]: I much prefer 2 spaces. but im also ok with whatever

**Pull request closed without merging**